mergify_engine/debug.py

```diff
@@ -157,19 +158,19 @@ async def report(
 ) -> typing.Union[context.Context, github.AsyncGithubInstallationClient, None]:
     redis_cache = utils.create_aredis_for_cache(max_idle_time=0)
 
-    path = url.replace("https://github.com/", "")
+    path = urllib.parse.urlparse(url).path
 
     pull_number: typing.Optional[str]
     repo: typing.Optional[str]
 
     try:
-        owner, repo, _, pull_number = path.split("/")
+        _, owner, repo, _, pull_number = path.split("/")
```

Comment:
This code is stil very fragile IMHO:
see:
```python
>>> urllib.parse.urlparse("https://github.com//foobar/lol/pull/12").path.split("/")
['', '', 'foobar', 'lol', 'pull', '12']
```

it's already broken in that regard, but it might be good to enhance it!